src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```diff
@@ -46,76 +51,108 @@ private enum Direction
+        public HttpCollectorWire(IConfiguration configuration, IAgentHealthReporter agentHealthReporter, IHttpClientFactory httpClientFactory)
         {
             _configuration = configuration;
             _requestHeadersMap = _emptyRequestHeadersMap;
             _agentHealthReporter = agentHealthReporter;
+            _httpClientFactory = httpClientFactory;
         }
 
-        public HttpCollectorWire(IConfiguration configuration, Dictionary<string, string> requestHeadersMap, IAgentHealthReporter agentHealthReporter)
+        public HttpCollectorWire(IConfiguration configuration, Dictionary<string, string> requestHeadersMap, IAgentHealthReporter agentHealthReporter, IHttpClientFactory httpClientFactory)
         {
             _configuration = configuration;
             _requestHeadersMap = requestHeadersMap ?? _emptyRequestHeadersMap;
             _agentHealthReporter = agentHealthReporter;
+            _httpClientFactory = httpClientFactory;
         }
 
         public string SendData(string method, ConnectionInfo connectionInfo, string serializedData, Guid requestGuid)
         {
             try
             {
                 var uri = GetUri(method, connectionInfo);
-                
+
                 Log.DebugFormat("Request({0}): About to Invoke \"{1}\" with : {2}", requestGuid, method, serializedData);
-                AuditLog(Direction.Sent, Source.InstrumentedApp, uri);
+                AuditLog(Direction.Sent, Source.InstrumentedApp, uri.ToString());
                 AuditLog(Direction.Sent, Source.InstrumentedApp, serializedData);
 
                 var bytes = new UTF8Encoding().GetBytes(serializedData);
                 var uncompressedByteCount = bytes.Length;
                 var requestPayload = GetRequestPayload(bytes);
-                var request = BuildRequest(uri, connectionInfo, requestPayload);
 
-                // Check serializedData length < MaxPayloadSizeInBytes before sending
-                // if > silently drop the payload for now, ideally want to be able split the data
-                // into chunks to send
                 if (requestPayload.Data.Length > _configuration.CollectorMaxPayloadSizeInBytes)
                 {
-                    // Log that the payload is being dropped. 
+                    // Log that the payload is being dropped
                     Log.ErrorFormat("Request({0}): Dropped large payload: size: {1}, max_payload_size_bytes={2}", requestGuid, requestPayload.Data.Length, _configuration.CollectorMaxPayloadSizeInBytes);
                     _agentHealthReporter.ReportSupportabilityPayloadsDroppeDueToMaxPayloadSizeLimit(method);
                     return "{}"; // mimics what is sent by NoOpCollectorWire
                 }
 
-                var response = SendRequest(request, requestPayload.Data);
+                var httpClient = _httpClientFactory.CreateClient(connectionInfo.Proxy);
```

Comment:
Since `HttpClient` implements `IDisposable`, this block (down to line 144 or so) should probably be wrapped in a `using()` to ensure the client gets disposed when it goes out of scope.

Reply:
Further, is it necessary to create a new client object for every Send?